nomad/structs/structs.go

```diff
@@ -2074,6 +2074,7 @@ type ServiceCheck struct {
 	Interval      time.Duration // Interval of the check
 	Timeout       time.Duration // Timeout of the response from the check before consul fails the check
 	InitialStatus string        // Initial status of the check
+	TLSSkipVerify bool          // Skip TLS verification when Protocol=https
```

Comment:
Add to API package, job spec, tests, diff. 
1) https://github.com/hashicorp/nomad/blob/master/api/tasks.go#L81
2) https://github.com/hashicorp/nomad/blob/master/jobspec/parse.go#L941
3) https://github.com/hashicorp/nomad/blob/master/command/agent/job_endpoint.go#L496